lib/Onyx.js

```diff
@@ -1043,10 +1062,12 @@ function merge(key, changes) {
                 // After that we merge the batched changes with the existing value
                 let modifiedData = applyMerge(existingValue, [batchedChanges]);
 
-                // For objects, the key for null values needs to be removed from the object to ensure the value will get removed from storage completely.
-                // On native, SQLite will remove top-level keys that are null. To be consistent, we remove them on web too.
-                if (!_.isArray(modifiedData) && _.isObject(modifiedData)) {
-                    modifiedData = _.omit(modifiedData, value => _.isNull(value));
+                modifiedData = cleanObject(modifiedData);
+
+                // On the first write SQLite doesn't use JSON_PATCH, which means, nullish values won't be omitted.
```

Comment:
How does this code know whether or not this is the first write or not? Also, why doesn't it use JSON_PATCH on the first write? Also, why does only JSON_PATCH remove null values?

I know that's a lot of questions, but answering them in the code comment will help other engineers that see this.